File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/UnityLevelPlay/LevelPlayInterstitialVC.m
#import "LevelPlayInterstitialVC.h"
#import <IronSource/IronSource.h>
#import "AdConfig.h"

static NSString *const TAG = @"LevelPlay-interstitial:";

@interface LevelPlayInterstitialVC () <LPMInterstitialAdDelegate>
@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, strong) LPMInterstitialAd *interstitialAd;
@property (nonatomic, strong) UILabel *label;
@end

@implementation LevelPlayInterstitialVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"levelPlay_interstitial", @"");

    UIButton *bnLoad = [self createButtonWithTitle:NSLocalizedString(@"load_ad", @"") action:@selector(buttonLoad)];
    [self.view addSubview:bnLoad];

    UIButton *bnShow = [self createButtonWithTitle:NSLocalizedString(@"show_ad", @"") action:@selector(buttonShow)];
    [self.view addSubview:bnShow];

    self.label = [self createLabel];
    [self.view addSubview:self.label];

    [NSLayoutConstraint activateConstraints:@[
        [bnLoad.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
        [bnLoad.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnLoad.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnLoad.heightAnchor constraintEqualToConstant:50],

        [bnShow.topAnchor constraintEqualToAnchor:bnLoad.bottomAnchor constant:20],
        [bnShow.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [bnShow.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [bnShow.heightAnchor constraintEqualToConstant:50],

        [self.label.topAnchor constraintEqualToAnchor:bnShow.bottomAnchor constant:20],
        [self.label.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
        [self.label.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
        [self.label.heightAnchor constraintEqualToConstant:50]
    ]];
}

- (UIButton *)createButtonWithTitle:(NSString *)title action:(SEL)action {
    UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
    [button setTitle:title forState:UIControlStateNormal];
    [button addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    button.translatesAutoresizingMaskIntoConstraints = NO;
    return button;
}

- (UILabel *)createLabel {
    UILabel *label = [[UILabel alloc] init];
    label.textAlignment = NSTextAlignmentCenter;
    label.translatesAutoresizingMaskIntoConstraints = NO;
    return label;
}

- (void)buttonLoad {
    NSLog(@"%@ buttonLoad", TAG);
    if (self.isLoading) return;
    [self updateUI:YES message:NSLocalizedString(@"loading", @"")];
    [self loadAd];
}

- (void)buttonShow {
    if (self.interstitialAd && [self.interstitialAd isAdReady]) {
        [self.interstitialAd showAdWithViewController:self placementName:nil];
    } else {
        NSLog(@"%@ Ad wasn't ready", TAG);
    }
}

- (void)loadAd {
    self.interstitialAd = [[LPMInterstitialAd alloc] initWithAdUnitId:AdConfig.LevelPlay_Interstitial_Ad_Id];
    [self.interstitialAd setDelegate:self];
    [self.interstitialAd loadAd];
}

- (void)updateUI:(BOOL)loading message:(NSString *)msg {
    self.isLoading = loading;
    self.label.text = msg;
}

#pragma mark - LPMInterstitialAdDelegate

- (void)didLoadAdWithAdInfo:(LPMAdInfo *)adInfo {
    NSLog(@"%@ didLoadAd", TAG);
    dispatch_async(dispatch_get_main_queue(), ^{
        [self updateUI:NO message:NSLocalizedString(@"load_success", @"")];
    });
}

- (void)didFailToLoadAdWithAdUnitId:(NSString *)adUnitId error:(NSError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.localizedDescription];
    NSLog(@"%@ didFailToLoadAd: %@", TAG, msg);
    dispatch_async(dispatch_get_main_queue(), ^{
        [self updateUI:NO message:[NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg]];
    });
}

- (void)didDisplayAdWithAdInfo:(LPMAdInfo *)adInfo {
    NSLog(@"%@ didDisplayAd", TAG);
}

- (void)didFailToDisplayAdWithAdInfo:(LPMAdInfo *)adInfo error:(NSError *)error {
    NSLog(@"%@ didFailToDisplayAd: %@", TAG, error.localizedDescription);
}

- (void)didClickAdWithAdInfo:(LPMAdInfo *)adInfo {
    NSLog(@"%@ didClickAd", TAG);
}

- (void)didCloseAdWithAdInfo:(LPMAdInfo *)adInfo {
    NSLog(@"%@ didCloseAd", TAG);
}

@end

File: SwiftPodDemo/AlxAdsDemo/UI/UnityLevelPlay/LevelPlayInterstitialVC.swift
//
//  LevelPlayInterstitialVC.swift
//  AlxAdsDemo
//
//  Created by YXk on 2026/4/1.
//

import UIKit
import IronSource

// MARK: - Unity LevelPlay Interstitial Ad
class LevelPlayInterstitialVC: BaseUIViewController {
    private let TAG = "LevelPlay-interstitial:"

    private var label: UILabel!
    private var isLoading = false
    private var interstitialAd: LPMInterstitialAd?

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
        navigationItem.title = NSLocalizedString("levelPlay_interstitial", comment: "")

        let bnLoad = createButton(title: NSLocalizedString("load_ad", comment: ""),
                                  action: #selector(buttonLoad))
        view.addSubview(bnLoad)

        let bnShow = createButton(title: NSLocalizedString("show_ad", comment: ""),
                                  action: #selector(buttonShow))
        view.addSubview(bnShow)

        label = createLabel()
        view.addSubview(label)

        NSLayoutConstraint.activate([
            bnLoad.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            bnLoad.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnLoad.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnLoad.heightAnchor.constraint(equalToConstant: 50),

            bnShow.topAnchor.constraint(equalTo: bnLoad.bottomAnchor, constant: 20),
            bnShow.leftAnchor.constraint(equalTo: view.leftAnchor),
            bnShow.rightAnchor.constraint(equalTo: view.rightAnchor),
            bnShow.heightAnchor.constraint(equalToConstant: 50),

            label.topAnchor.constraint(equalTo: bnShow.bottomAnchor, constant: 20),
            label.leftAnchor.constraint(equalTo: view.leftAnchor),
            label.rightAnchor.constraint(equalTo: view.rightAnchor),
            label.heightAnchor.constraint(equalToConstant: 50),
        ])
    }

    @objc private func buttonLoad() {
        print("\(TAG) buttonLoad")
        guard !isLoading else { return }
        updateUI(true, NSLocalizedString("loading", comment: ""))
        loadAd()
    }

    @objc private func buttonShow() {
        guard let ad = interstitialAd, ad.isAdReady() else {
            print("\(TAG) Ad wasn't ready")
            return
        }
        ad.showAd(viewController: self, placementName: nil)
    }

    private func loadAd() {
        interstitialAd = LPMInterstitialAd(adUnitId: AdConfig.LevelPlay_Interstitial_Ad_Id)
        interstitialAd?.setDelegate(self)
        interstitialAd?.loadAd()
    }

    private func updateUI(_ loading: Bool, _ msg: String) {
        isLoading = loading
        label.text = msg
    }
}

// MARK: - LPMInterstitialAdDelegate

extension LevelPlayInterstitialVC: LPMInterstitialAdDelegate {

    func didLoadAd(with adInfo: LPMAdInfo) {
        print("\(TAG) didLoadAd")
        DispatchQueue.main.async { [weak self] in
            self?.updateUI(false, NSLocalizedString("load_success", comment: ""))
        }
    }

    func didFailToLoadAd(withAdUnitId adUnitId: String, error: Error) {
        let msg = "\((error as NSError).code): \(error.localizedDescription)"
        print("\(TAG) didFailToLoadAd: \(msg)")
        DispatchQueue.main.async { [weak self] in
            self?.updateUI(false, String(format: NSLocalizedString("load_failed", comment: ""), msg))
        }
    }

    func didDisplayAd(with adInfo: LPMAdInfo) {
        print("\(TAG) didDisplayAd")
    }

    func didFailToDisplayAd(with adInfo: LPMAdInfo, error: Error) {
        print("\(TAG) didFailToDisplayAd: \(error.localizedDescription)")
    }

    func didClickAd(with adInfo: LPMAdInfo) {
        print("\(TAG) didClickAd")
    }

    func didCloseAd(with adInfo: LPMAdInfo) {
        print("\(TAG) didCloseAd")
    }
}
